source/agora/consensus/data/Transaction.d

```diff
@@ -129,15 +129,26 @@ public struct Output
 /// The input of the transaction, which spends a previously received `Output`
 public struct Input
 {
-    /// The hash of a previous transaction containing the `Output` to spend
-    public Hash previous;
-
-    /// Index of the `Output` in the `previous` `Transaction`
-    public uint index;
+    /// The hash of the UTXO to be spent
+    public Hash utxo;
 
     /// A signature that should be verified using the `previous[index].address` public key
     public Signature signature;
 
+    /// Simple ctor
+    public this (in Hash utxo_, in Signature sig = Signature.init)
```

Comment:
Wouldn't it make sense to just remove the `Signature` parameter from the ctor?

I'm not sure we can sign anything before the utxo is set?

Reply:
Well since the signature is on the `Hash`, we can.
And the reason this ctor is here is because the deserializer needs it.

Reply:
It's on the hash of the transaction, is it not?

But in any case this stuff might change in the future (SIGHASH, etc). So no worries for now.

Reply:
Ah right! Still, if you remove it, it doesn't compile 🤷